test/DynamoCoreTests/CoreTests.cs

```diff
@@ -83,6 +83,45 @@ public void CanAddAnnotation()
 
         [Test]
         [Category("UnitTests")]
+        public void UndoAnnotationText()
+        {
+            //Add a Node
+            var model = ViewModel.Model;
+            var addNode = new DSFunction(model.LibraryServices.GetFunctionDescriptor("+"));
+            model.CurrentWorkspace.AddNode(addNode, false);
+            Assert.AreEqual(ViewModel.CurrentSpace.Nodes.Count, 1);
+
+            //Add a Note 
+            Guid id = Guid.NewGuid();
+            var addNote = ViewModel.Model.CurrentWorkspace.AddNote(false, 200, 200, "This is a test note", id);
+            Assert.AreEqual(ViewModel.CurrentSpace.Notes.Count, 1);
+
+            //Select the node and notes
+            DynamoSelection.Instance.Selection.Add(addNode);
+            DynamoSelection.Instance.Selection.Add(addNote);
+
+            //create the group around selected nodes and notes
+            Guid groupid = Guid.NewGuid();
+            var annotation = ViewModel.Model.CurrentWorkspace.AddAnnotation("This is a test group", groupid);
+            Assert.AreEqual(ViewModel.CurrentSpace.Annotations.Count, 1);
+            Assert.AreNotEqual(0, annotation.Width);
+
+            //Update the Annotation Text
+            ViewModel.ExecuteCommand(
+                    new DynCmd.UpdateModelValueCommand(
+                        System.Guid.Empty, annotation.GUID, "TextBlockText",
+                        "This is a unit test"));
+            Assert.AreEqual("This is a unit test", annotation.AnnotationText);
+
+            //Undo Annotation text
+            ViewModel.UndoCommand.Execute(null);
+
+            //Title should be changed now.
+            Assert.AreEqual("This is a test group", annotation.AnnotationText);
+        }
+
```

Comment:
Is it possible for this test to inherit from `DynamoModelUnitTestBase`?  Then we don't need to construct a `ViewModel` layer.  Beyond removing a bunch of code-noise (e.g. `Model` as opposed to `ViewModel.Model`) this means your test can actually be executed on Linux or Mac.